snapcraft/file_utils.py

```diff
@@ -150,18 +150,22 @@ def _search_and_replace_contents(file_path, search_pattern, replacement):
     if os.path.islink(file_path):
         return
 
-    with open(file_path, 'r+') as f:
-        try:
-            original = f.read()
-        except UnicodeDecodeError:
-            # This was probably a binary file. Skip it.
-            return
-
-        replaced = search_pattern.sub(replacement, original)
-        if replaced != original:
-            f.seek(0)
-            f.truncate()
-            f.write(replaced)
+    try:
+        with open(file_path, 'r+') as f:
+            try:
+                original = f.read()
+            except UnicodeDecodeError:
+                # This was probably a binary file. Skip it.
+                return
+
+            replaced = search_pattern.sub(replacement, original)
+            if replaced != original:
+                f.seek(0)
+                f.truncate()
+                f.write(replaced)
+    except PermissionError as e:
+        logger.debug('Unable to open {path} for writing: {error}'.format(
+            path=file_path, error=e))
```

Comment:
Sorry for the late feedback here.
I don't like this huge try-catch.  I think that the statement that would throw PermissionError is f.truncate, isn't it?
If so, we can put a smaller try catch at the end of this code block.

Also, an alternative would be to check if the file is writable before trying to write it.